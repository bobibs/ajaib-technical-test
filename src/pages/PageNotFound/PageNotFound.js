import React from 'react';
import { useNavigate } from 'react-router-dom';
import Button from '../../components/elements/Button';
import ImageBox from '../../components/elements/ImageBox';
import styles from './styles.module.css';

export default function PageNotFound() {
  const navigate = useNavigate();
  const handleClick = () => navigate('/');

  return (
    <div className={styles.root}>
      <ImageBox className={styles.image} src='404' />
      <Button onClick={handleClick} size='small' variant='outlined'>
        Back to home
      </Button>
    </div>
  );
}
