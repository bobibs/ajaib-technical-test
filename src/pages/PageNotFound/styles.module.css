.root {
    height: 100vh;
    background-color: var(--primary-color);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.image {
    height: 10rem;
    margin: 0 0 2rem 0;
}
