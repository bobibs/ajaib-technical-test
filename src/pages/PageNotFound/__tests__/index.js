import React from 'react';
import { useNavigate } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageNotFound from '../PageNotFound';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn((fn) => fn()),
}));

const navigate = jest.fn();

describe('src/components/elements/PageNotFound', () => {
  beforeEach(() => {
    useNavigate.mockImplementation(() => navigate);
  });

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PageNotFound />);
    expect(tree).toMatchSnapshot();
  });

  test('function handleClick', () => {
    const result = PageNotFound();
    result.props.children[1].props.onClick();
    expect(navigate).toHaveBeenCalled();
  });
});
