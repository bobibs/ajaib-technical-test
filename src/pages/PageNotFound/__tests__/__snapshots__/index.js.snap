// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/PageNotFound render 1`] = `
<div
  className="root"
>
  <ImageBox
    className="image"
    format="svg"
    onClick={[Function]}
    src="404"
  />
  <Button
    className=""
    onClick={[Function]}
    size="small"
    variant="outlined"
  >
    Back to home
  </Button>
</div>
`;
